### PR TITLE
Update GHA matrix template

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Update GHA matrix template.

Relates to https://github.com/ansible-community/community-topics/issues/237

### CODE REVIEW

**Comment by Andersson007** [comment]
cc @felixfontein @briantist 
thanks @Spredzy for mentioning this yesterday

File: .github/workflows/ansible-test.yml [hunk 1]
```diff
@@ -185,72 +193,10 @@ jobs:
...
-          #   python: '3.8'
-          # - ansible: stable-2.12
-          #   python: '3.9'
-          # - ansible: stable-2.12
-          #   python: '3.10'
-          # ansible-core 2.13
+          # Timely add new versions announced in
+          # https://github.com/ansible-collections/news-for-maintainers,
+          # consider dropping testing against EOL versions.
+          # ansible-core 2.14
```
Comment (felixfontein) [inline .github/workflows/ansible-test.yml:199]:
```suggestion
          # ansible-core 2.13
```

Comment (Andersson007) [reply .github/workflows/ansible-test.yml]: Good eye, thanks:) committed, PTAL

**Review by felixfontein — APPROVED** [review]

**Comment by briantist** [comment]
I feel somewhat uncomfortable about asking (especially new) maintainers to try to limit their CI usage. I think the decision of what to test should be primarily based on what makes for good content and test coverage, and I worry this sends a message to folks that they should be spending more time and effort on cutting tests.

I do think it's a good idea to explain that limits on concurrency specifically are shared limits in the organization and to be mindful especially of long-running tests... but I do want to be careful about the wording.

I'm not sure if I'm overthinking the message.

(comment also applies to https://github.com/ansible/ansible-documentation/pull/40)

**Comment by Andersson007** [comment]
> I feel somewhat uncomfortable about asking (especially new) maintainers to try to limit their CI usage. I think the decision of what to test should be primarily based on what makes for good content and test coverage, and I worry this sends a message to folks that they should be spending more time and effort on cutting tests.
> 
> I do think it's a good idea to explain that limits on concurrency specifically are shared limits in the organization and to be mindful especially of long-running tests... but I do want to be careful about the wording.
> 
> I'm not sure if I'm overthinking the message.
> 
> (comment also applies to [ansible/ansible-documentation#40](https://github.com/ansible/ansible-documentation/pull/40))

@briantist sounds fair, thanks! Changed the wording a bit. How about now?

File: .github/workflows/ansible-test.yml [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+#    (click the Watch button on the homepage > Custom > Issues)
+#    and keep this matrix up to date in accordance to related announcements.
+#    Timely add new ansible-core versions and consider dropping support
+#    and testing against its EOL versions.
+# 2. If your collection repository is under the ansible-collections org,
+#    please keep in mind that the number of GHA jobs is limited
+#    and shared across all the collections in the org.
+#    So, focusing on good test coverage of your collection,
+#    please avoid testing against unnecessary entities such as
+#    ansible-core EOL versions your collection does not support.
```
Comment (mariolenz) [inline .github/workflows/ansible-test.yml:12]:
```suggestion
#    ansible-core EOL versions or versions your collection does not support.
```

Avoid testing against ansible-core versions that are _not_ EOL yet but not supported by the collection helps, too.

File: .github/workflows/ansible-test.yml [hunk 3]
```diff
@@ -185,71 +195,9 @@ jobs:
...
-          #   python: '3.7'
-          # - ansible: stable-2.12
-          #   python: '3.8'
-          # - ansible: stable-2.12
-          #   python: '3.9'
-          # - ansible: stable-2.12
-          #   python: '3.10'
+          # Timely add new versions announced in
+          # https://github.com/ansible-collections/news-for-maintainers,
+          # consider dropping testing against EOL versions.
```
Comment (mariolenz) [inline .github/workflows/ansible-test.yml:200]:
```suggestion
          # consider dropping testing against EOL versions and versions you don't officially support.
```

Similar to my comment above.

**Review by mariolenz — COMMENTED** [review]
Just some ideas, don't know if they are helpful.

**Review by russoz — APPROVED** [review]

**Comment by Andersson007** [comment]
@felixfontein @briantist @mariolenz @russoz thanks a lot for reviewing!

**Pull request merged**